File: src/main/java/com/github/zoewithabang/model/Alias.java
package com.github.zoewithabang.model;

import java.util.Objects;

public class Alias implements IModel
{
    private String alias;
    private String command;
    private String description;
    
    public Alias(){}
    
    public Alias(String alias, String command, String description)
    {
        this.alias = alias;
        this.command = command;
        this.description = description;
    }
    
    public String getAlias()
    {
        return alias;
    }
    
    public void setAlias(String alias)
    {
        this.alias = alias;
    }
    
    public String getCommand()
    {
        return command;
    }
    
    public void setCommand(String command)
    {
        this.command = command;
    }
    
    public String getDescription()
    {
        return description;
    }
    
    public void setDescription(String description)
    {
        this.description = description;
    }
    
    @Override
    public String toString()
    {
        return "Alias{" +
            "alias='" + alias + '\'' +
            ", command='" + command + '\'' +
            ", description='" + description + '\'' +
            '}';
    }
    
    @Override
    public boolean equals(Object o)
    {
        if(this == o) return true;
        if(o == null || getClass() != o.getClass()) return false;
        Alias alias1 = (Alias) o;
        return Objects.equals(alias, alias1.alias) &&
            Objects.equals(command, alias1.command) &&
            Objects.equals(description, alias1.description);
    }
    
    @Override
    public int hashCode()
    {
        return Objects.hash(alias, command, description);
    }
}

File: src/main/kotlin/com/github/zoewithabang/model/Alias.kt
package com.github.zoewithabang.model

class Alias {
    var alias: String? = null
    var command: String? = null
    var description: String? = null

    constructor()

    constructor(alias: String, command: String, description: String) {
        this.alias = alias
        this.command = command
        this.description = description
    }

    override fun toString(): String {
        return "Alias(alias=$alias, command=$command, description=$description)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Alias

        if (alias != other.alias) return false
        if (command != other.command) return false
        if (description != other.description) return false

        return true
    }

    override fun hashCode(): Int {
        var result = alias?.hashCode() ?: 0
        result = 31 * result + (command?.hashCode() ?: 0)
        result = 31 * result + (description?.hashCode() ?: 0)
        return result
    }
}
